flink-core/src/main/java/org/apache/flink/core/plugin/PluginConfig.java

```diff
@@ -55,23 +55,22 @@ private PluginConfig(Optional<Path> pluginsPath, String[] alwaysParentFirstPatte
 
 	public static PluginConfig fromConfiguration(Configuration configuration) {
 		return new PluginConfig(
-			getPluginsDirPath(configuration),
+			getPluginsDir().map(File::toPath),
 			CoreOptions.getParentFirstLoaderPatterns(configuration));
 	}
 
-	private static Optional<Path> getPluginsDirPath(Configuration configuration) {
-		String pluginsDir = configuration.getString(ConfigConstants.ENV_FLINK_PLUGINS_DIR, null);
-		if (pluginsDir == null) {
-			LOG.info("Environment variable [{}] is not set", ConfigConstants.ENV_FLINK_PLUGINS_DIR);
-			return Optional.empty();
-		}
+	public static Optional<File> getPluginsDir() {
+		String pluginsDir = System.getenv().getOrDefault(
+			ConfigConstants.ENV_FLINK_PLUGINS_DIR,
+			ConfigConstants.DEFAULT_FLINK_PLUGINS_DIRS);
+
 		File pluginsDirFile = new File(pluginsDir);
 		if (!pluginsDirFile.isDirectory()) {
 			LOG.warn("Environment variable [{}] is set to [{}] but the directory doesn't exist",
```

Comment:
If the env is not set and DEFAULT_FLINK_PLUGINS_DIRS does not exist, it will show a confusing log. 
`Environment variable [FLINK_PLUGINS_DIR] is set to [plugins] but the directory doesn't exist`

Reply:
True. Let me change it into "The plugins directory [plugins] does not exist."